### PR TITLE
Support string in test component, require and tag

### PR DESCRIPTION
**Pull request by psss**

### CODE REVIEW

File: spec/tests/component.fmf [hunk 1]
```diff
@@ -10,10 +10,16 @@ description:
     given component or package. As they do not always have to be
     stored in the same repository and because many tests cover
     multiple components a dedicated field is needed. Should be a
-    ``list of strings``.
+    ``string`` or a ``list of strings``. Component name usually
+    corresponds to the source package name.
 
 example: |
+    component: libselinux checkpolicy
```
Comment (thrix) [inline spec/tests/component.fmf:17]: This is not a standard notion, if we enable it like this here, we should document it explicitly here or somewhere else.

Comment (psss) [reply spec/tests/component.fmf:17]: Yeah, let's discuss this today on the hacking session. Either we remove support for splitting on white-space/comma or we clearly document it.

Comment (thrix) [reply spec/tests/component.fmf:17]: yeah, definitely

Comment (psss) [reply spec/tests/component.fmf]: Split on white-space/comma removed.

File: tests/unit/test_base.py [hunk 2]
```diff
@@ -50,6 +50,10 @@ def test_test_invalid():
     # Invalid attributes
     for key in ['component', 'require', 'tag']:
         with pytest.raises(tmt.utils.SpecificationError):
-            test = tmt.Test({key: 'string'}, name='/smoke')
+            test = tmt.Test({key: 1}, name='/smoke')
     with pytest.raises(tmt.utils.SpecificationError):
         test = tmt.Test({'environment': 'string'}, name='/smoke')
+    # Listify attributes
+    assert tmt.Test({'tag': 'a'}, name='/smoke').tag == ['a']
+    assert tmt.Test({'tag': 'a b c'}, name='/smoke').tag == ['a b c']
```
Comment (lukaszachy) [inline tests/unit/test_base.py:58]: I understand this is test of listify, however per documentation input value is invalid (comma, space separation). Maybe to pick another attribute (where spaces and commas are allowed)?

Comment (psss) [reply tests/unit/test_base.py:58]: Right your are, now it does not make sense here at all. Removing it as `listify` has its own coverage.

File: spec/tests/component.fmf (hunk 1 above)
[the remark this answers is not in the document]
Comment (psss) [reply spec/tests/component.fmf]: White-space/comma split removed.

File: tmt/utils.py [hunk 3]
```diff
@@ -459,12 +459,15 @@ def listify(data, split=False, keys=None):
     Ensure that variable is a list, convert if necessary
 
     For dictionaries check all items or only those with provided keys.
-    Also split strings on space/comma if split=True.
+    Also split strings on white-space/comma if split=True.
```
[the remark this answers is not in the document]
Comment (psss) [reply tmt/utils.py:462]: White-space/comma split removed.

**Pull request merged**